BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,364 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import Shared
+
+struct NetworkInputItem {
+  var input: String
+  var error: String?
+}
+
+struct NetworkTextField: View {
+  var placeholder: String
+  @Binding var item: NetworkInputItem
+  var onChange: (String) -> Void
+  
+  var body: some View {
+    VStack(alignment: .leading) {
+      TextField(placeholder, text: $item.input)
+        .onChange(of: item.input) { newValue in
+          onChange(newValue)
+        }
+        .autocapitalization(.none)
+        .disableAutocorrection(true)
+      if let error = item.error {
+        HStack(alignment: .firstTextBaseline, spacing: 4) {
+          Image(systemName: "exclamationmark.circle.fill")
+          Text(error)
+            .fixedSize(horizontal: false, vertical: true)
+              .animation(nil, value: error)
+        }
+        .transition(
+          .asymmetric(
+            insertion: .opacity.animation(.default),
+            removal: .identity
+          )
+        )
+        .font(.footnote)
+        .foregroundColor(Color(.braveErrorLabel))
+      }
+    }
+  }
+}
+
+class CustomNetworkModel: ObservableObject {
+  @Published var networkId = NetworkInputItem(input: "")
+  @Published var networkName = NetworkInputItem(input: "")
+  @Published var networkSymbolName = NetworkInputItem(input: "")
+  @Published var networkSymbol = NetworkInputItem(input: "")
+  @Published var networkDecimals = NetworkInputItem(input: "")
+  
+  @Published var rpcUrls: [NetworkInputItem] = [NetworkInputItem(input: "")]
+  @Published var iconUrls = [NetworkInputItem(input: "")]
+  @Published var blockUrls = [NetworkInputItem(input: "")]
+  
+  @Published var testUrls = [NetworkInputItem(input: "")]
+  
+  init(network: BraveWallet.EthereumChain? = nil) {
```

Comment:
hmmmm this model will take care of textFields input default values for both edit(with existed network info as default values for input fields) and non-edit mode which all fields has default value with empty input or empty list.
Does `init(with network: BraveWallet.EthereumChain? = nil)` make it better to understand.

Reply:
@nuo-xu The main part I don't like about this pattern is you're taking an optional in order to do conditional logic _within_ the function rather than at the call point. Like this argument shouldn't be optional if it does _nothing at all_ when its `nil`. Instead it should be up to the place calling it to not call the function when that argument would be `nil`.  What makes this more confusing is that not only is this an initializer, its the only initializer used and its fully dependent on the default values.

Something less confusing would be using `init()` to create it and setting up the values from a network elsewhere.  If you want to keep that logic bottled up within this class, you could do something along the lines of:

```swift 
/// Updates the details of this class based on a custom network
func populateDetails(from network: BraveWallet.EthereumChain) {
   ...
}
```

or

```swift
/// Creates a details object based on a custom network
static func details(from network: BraveWallet..EthereumChain) {
   ...
}
```

Then at call site you just make a details object and if network is non-nil you call the appropriate method.  Personally I like the `static func` but it means you'll have to do something 

```swift
var details: CustomDetailsObject
if let network = network {
  details = .details(from: network)
} else {
  details = .init()
}
```